File: js/log/log.go
package log

import (
	log "github.com/Sirupsen/logrus"
)

func Log(logger *log.Logger, t, msg string, fields map[string]interface{}) {
	e := logger.WithFields(log.Fields(fields))
	switch t {
	case "error":
		e.Error(msg)
	case "warn":
		e.Warn(msg)
	case "info":
		e.Info(msg)
	case "debug":
		e.Debug(msg)
	}
}

File: js/lib/api/log.js
"use strict";

$log = $log || {};

$log.debug = function() {
	return $log.log.apply(this, _.concat(['debug'], arguments));
}

$log.info = function() {
	return $log.log.apply(this, _.concat(['info'], arguments));
}

$log.warn = function() {
	return $log.log.apply(this, _.concat(['warn'], arguments));
}

$log.warning = function() {
	return $log.warn.apply(this, arguments);
}

$log.error = function() {
	return $log.log.apply(this, _.concat(['error'], arguments));
}
